### PR TITLE
refactor(config): Replace config file with env variables

### PR DESCRIPTION
**Pull request by bcwaldon**

### CODE REVIEW

File: app/config.go [hunk 1]
```diff
@@ -25,17 +24,36 @@ func (cfg *Config) ListenPort() string {
...
+	for _, e := range env {
+		tokens := strings.SplitN(e, "=", 2)
+		vars[tokens[0]] = tokens[1]
 	}
 
-	var cfg Config
-	err = json.Unmarshal(configBytes, &cfg)
-	if err != nil {
-		panic(err)
+	listen, ok := vars["Subscribegun_Listen"]
```
Comment (philips) [inline app/config.go]: SUBGUN_LISTEN

File: app/config.go [hunk 2]
```diff
@@ -25,17 +24,36 @@ func (cfg *Config) ListenPort() string {
...
+	if !ok {
+		return nil, errors.New("Environment variable Subscribegun_Listen could not be found")
+	}
+	cfg.Subscribegun.Listen = listen
+
+	listCSV, ok := vars["Subscribegun_Lists"]
```
Comment (philips) [inline app/config.go]: SUBGUN_LISTS

File: app/config.go [hunk 3]
```diff
@@ -25,17 +24,36 @@ func (cfg *Config) ListenPort() string {
...
+		return nil, errors.New("Environment variable Subscribegun_Lists could not be found")
 	}
-	return &cfg
+	lists := strings.Split(listCSV, ",")
+	if len(lists) == 0 {
+		return nil, errors.New("Environment variable Subscribegun_Lists provides no lists")
+	}
+	cfg.Subscribegun.Lists = lists
+
+	key, ok := vars["Mailgun_Key"]
```
Comment (philips) [inline app/config.go]: SUBGUN_MAILGUN_API_KEY?

Comment (bcwaldon) [inline app/config.go]: SUBGUN_API_KEY

**Comment by bcwaldon** [comment]
updated env var names

**Comment by philips** [comment]
MOAR_TESTS_REQUIRED

**Comment by bcwaldon** [comment]
@philips tests added!

**Comment by philips** [comment]
lgtm, I am going to merge this.

**Comment by philips** [comment]
@bcwaldon Oh, we should update the README in this PR.

**Comment by bcwaldon** [comment]
@philips Updated README.md & Dockerfile; dropped config.json.tmpl

**Pull request merged**